core/src/components/datetime/datetime.scss

```diff
@@ -32,24 +32,37 @@ ion-picker-column-internal {
   min-width: 26px;
 }
 
-:host(.datetime-size-fixed) {
-  width: auto;
-  height: auto;
-}
-
-:host(.datetime-size-fixed:not(.datetime-prefer-wheel)) {
-  max-width: 350px;
+/**
+ * This sets the maximum size of the wheel picker.
+ */
+:host(.datetime-size-fixed:not(.datetime-grid)) {
+  max-width: #{$datetime-fixed-width};
 }
 
 /**
- * This ensures that the picker is apppropriately
- * sized and never truncates the text.
+ * The size of the wheel picker
+ * with date, date-time, or time-date
+ * is much larger than the wheel picker
+ * with just time or month, so we make adjustments
+ * here so the sizing is appropriate.
  */
 :host(.datetime-size-fixed.datetime-prefer-wheel) {
-  min-width: 350px;
+  min-width: #{$datetime-fixed-width};
   max-width: max-content;
 }
 
+/**
+ * When using the datetime grid interface,
+ * we want the component to keep the same
+ * width even when the month/year picker is open.
+ * If we do not set the width here then
+ * the component will cause a layout shift
+ * when the month/year picker opens.
+ */
+:host(.datetime-size-fixed.datetime-grid) {
+  width: #{$datetime-fixed-width};
```

Comment:
Many of the screenshot updates that came from this change have blanked out calendars (ex. `core/src/components/datetime/test/color/datetime.e2e.ts-snapshots/datetime-color-custom-dark-color-md-ltr-Mobile-Firefox-linux.png`). We may need an `await page.waitForSelector('.datetime-ready')` somewhere in the test(s), or some other tweak to the `ionRender` logic.